scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -73,7 +76,10 @@ def _decode(self, body, encoding):
                 # http://www.gzip.org/zlib/zlib_faq.html#faq38
                 body = zlib.decompress(body, -15)
         if encoding == b'br' and b'br' in ACCEPTED_ENCODINGS:
-            body = brotli.decompress(body)
+            if (flag):
```

Comment:
The `b'br' in ACCEPTED_ENCODINGS` in the `if` statement above already implies that `flag` is `0` here. So `flag` is unnecessary here.

Reply:
I think the condition here needs to be
```
if encoding == b'br':
    if b'br' in ACCEPTED_ENCODINGS:
        body = brotli.decompress(body)
    else:
        # warning or error
```

Reply:
Also, I believe we'll need a test for this, to ensure we get the message when required